tutorials/shaders/shader_reference/shader_preprocessor.rst

```diff
@@ -0,0 +1,141 @@
+.. _doc_shader_preprocessor:
+
+Shader Preprocessor
+===================
+
+Shader preprocessor belongs as optional step before shader compilation of text shaders in Godot.
```

Comment:
```suggestion
The shader preprocessor is an optional step before shader compilation of text shaders in Godot.
```
Also not sure what `text shaders` here means? I assume "before compiling shader code (text)", but IMO this wording is confusing, sounds like it is about a text/label specific shader thing.

Reply:
The shaders are divided to either text or visual, that what I mean.

Optional means user may not ever use the shader preprocessor if it doesn't need. Probably not a good sentence.